pipeline/container.go

```diff
@@ -189,31 +194,46 @@ func (c *Container) Execute(r *RuleData) bool {
 		// disregard the need to run the container
 		execute = false
 
+		match, err := c.Ruleset.Match(r)
+		if err != nil {
+			return false, err
+		}
+
 		// check if you need to run a status failure ruleset
 
 		if ((!(c.Ruleset.If.Empty() && c.Ruleset.Unless.Empty()) &&
 			!(c.Ruleset.If.NoStatus() && c.Ruleset.Unless.NoStatus())) || c.Ruleset.If.Parallel) &&
-			c.Ruleset.Match(r) {
+			match {
 			// approve the need to run the container
 			execute = true
 		}
 	}
 
 	r.Status = constants.StatusFailure
 
+	match, err := c.Ruleset.Match(r)
+	if err != nil {
+		return false, err
+	}
+
 	// check if you need to skip a status failure ruleset
 	if strings.EqualFold(status, constants.StatusSuccess) &&
 		!(c.Ruleset.If.NoStatus() && c.Ruleset.Unless.NoStatus()) &&
-		!(c.Ruleset.If.Empty() && c.Ruleset.Unless.Empty()) && c.Ruleset.Match(r) {
+		!(c.Ruleset.If.Empty() && c.Ruleset.Unless.Empty()) && match {
```

Comment:
any concerns that .Match will be called for sure now whereas in the previous version other checks had to pass before it would _maybe_ call .Match?

Reply:
sorry, maybe i should have phrased it differently. i was trying to say that .Match is a bit more intensive to call than the other checks before it. now we are calling the more intensive things for sure. the difference might be negligible, however. just pointing it out :D